doc/aaa/SONiC RBAC.md

```diff
@@ -0,0 +1,486 @@
+    * *resource* : The xpath associated with the resource being accessed. This is a string.
+    * *feature-name* : The tag of the feature this resource belongs to.
+  The ResourceTable is keyed on <***resource, feature***>
+
+* **TenantTable**
+  (To be implemented in Phase 3 or when Multi-tenancy is introduced)
+  In most systems today, a single SONiC system will serve multiple tenants. A tenant is a group of users who have a different privileges for resource instances. As SONiC becomes multitenant, RBAC needs to account for this when authorizing users. The TenantTable is needed to enable this and has the following columns :
+    * *resource* : The xpath associated with the resource being accessed. This is a string.
+    * *tenant* : The tenant for which the resource partitioning is being done. This is a string.
+    * *instances* : The instances of the *resource* allocated to this *tenant*. This is a list of instances.
+  The TenantTable is keyed on <***resource, tenant***>
+
+### 3.2.2 APP DB
+N/A
+
+### 3.2.3 STATE DB
+N/A
+
+### 3.2.4 ASIC DB
+N/A
+
+### 3.2.5 COUNTER DB
+N/A
+
+## 3.3 Switch State Service Design
+### 3.3.1 Orchestration Agent
+N/A
+
+### 3.3.2 Other Process
+N/A
+
+## 3.4 SyncD
+
+N/A. HAM is going to be the single source of truth, and it will take care of
+synchronizing the User Table and /etc/passwd
+
+## 3.5 SAI
+N/A
+
+## 3.6 User Interface
+### 3.6.1 Data Models
+
+TBD from developer
+(TODO/DELL)
+
+### 3.6.2 CLI
+#### 3.6.2.1 Configuration Commands for User Management
+Users may be managed via Linux tools like `sonic-useradd`, `sonic-usermod`, `passwd`, etc. They may also be managed via configuration.
+
+##### username
+`username <name> password <password-string> role <role-string>` -- Configures a user on the system with a given name, password, and role.
+* **name** is a text string of 1-32 alphanumeric characters
+* **password-string** is a text string of 1-32 alphanumeric characters
+* **role-string** is a text string consisting of a role name. In the initial release, the user is recommended to use "admin" and "operator" roles, as other roles will not be supported. A text string is desired instead of keywords so that in the future, more roles may be implemented and expanded.
+* Configuring another a user with the same **name** should result in modification of the existing user.
+
+`no username <name>` -- Deletes a user from the system.
+* **name** is a text string of 1-32 alphanumeric characters
+
+##### aaa authentication
+```
+aaa authentication {rest-server | gnmi-server} {[password] [token] [certificate]}
```

Comment:
Will this need to have the cisco style "default" tag after {rest-server | gnmi-server}
`aaa authentication {rest-server | gnmi-server} default {[password][token][certificate]}`
or in this case there is no chance of needing a different authentication for KLISH connection over unix-domain socket vs remote clients connecting over TCP/TLS?